octodns/provider/route53.py

```diff
@@ -659,7 +661,19 @@ def r53_zones(self):
 
     def _get_zone_id(self, name, create=False):
         self.log.debug('_get_zone_id: name=%s', name)
-        if name in self.r53_zones:
+        if self.get_zones_by_name:
+            # attempt to get zone by name
+            # limited to one as this should be unique
+            response = self._conn.list_hosted_zones_by_name(
+                DNSName=name, MaxItems="1"
+            )
+            if len(response['HostedZones']) != 0:
+                # if there is a response that starts with the name
+                if response['HostedZones'][0]['Name'].startswith(name):
+                    id = response['HostedZones'][0]['Id']
+                    self.log.debug('get_zones_by_name:   id=%s', id)
+                    return id
+        elif name in self.r53_zones:
```

Comment:
Looks like `_get_zone_id` is called a few times during the course of planning. I believe this will result in calling `list_hosted_zones_by_name` a few times for each zone, quick scan looks like 3. If this is working for you that's probably ok, but it'll be a bit slower and more likely to hit rate limits. 

If this is working well enough I'm ok with it as-is. If you'd like to look into solving that I think the best option may be to remove `@property` from `r53_zone` and add a `name` param. That should allow the cache to still work and decide whether to get all or fetch just what it's asked for, something like:

```python
    def r53_zone(self, name):
        if self._r53_zones is None:
            if self. get_zones_by_name:
                self._r53_zones = {}
            else:
                ... existing implementation ...
        zone_id = self._r53_zones.get(name, None)
        if not zone_id and self.get_zones_by_name:
            ... new lookup by name ...
            self._r53_zones[name] = zone_id
        return zone_id
```

I think w/something like that `def _get_zone_id` could remain as it was. Lmk which way you'd like to go.

Reply:
This is a good point, I was originally torn on how I wanted to handle this and was avoiding removing this property on the r53_zone function. 

On the latest commit, I rewrote the r53_zone function to instead be an update function which updates the _r53_zones dictionary and handled the different cases where the dictionary would need to be updated. It's now functioning as 1 check per zone if using get_zones_by_name.